crates/fuel-core/src/service/genesis.rs

```diff
@@ -154,31 +171,67 @@ pub async fn execute_and_commit_genesis_block(
 }
 
 pub fn create_genesis_block(config: &Config) -> Block {
-    let block_height = config.snapshot_reader.block_height();
-    let da_block_height = config.snapshot_reader.da_block_height();
-    let transactions = vec![];
+    let height;
+    let da_height;
+    let consensus_parameters_version;
+    let state_transition_bytecode_version;
+
+    // If the rollup continues the old rollup, the height of the new block should
+    // be higher than that of the old chain by one to make it continuous.
+    // The same applies to the state transition functions and consensus
+    // parameters since it is a new chain.
+    if let Some(latest_block) = config.snapshot_reader.last_block_config() {
```

Comment:
It would be good to have some tracing on which branch it takes here.

Reply:
Added tracing into the `execute_genesis_block` function since it is the place where the choice matter